### PR TITLE
Add messaging for more than 2 complete references

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

Applications can get into a state where they have more than 2 references. We need to ask candidates to delete extra references, so that they can send it on to providers.

## Changes proposed in this pull request

- Reference review page updated to show an error when there are more than two refs in the `feedback_provided` state
- Application review page updated to:
  - show warning on the references section when in this state
  - show error on the references section on application submit

### Reference review page:
---
Screenshot_20201204_091757

### Application review page (warning):
---
Screenshot_20201204_091856

### Application review page (error):
---
Screenshot_20201204_091928

## Guidance to review

- In contrast to the card description, I've agreed with @paulrobertlloyd to try out the same messaging on both review pages to see how it looks, as the implementation is simpler.
- Per-commit review recommended.

## Link to Trello card
https://trello.com/c/IIWPdP7w

## Things to check

- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] API release notes have been updated if necessary
- [ ] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/candidate_interface/references/review/show.html.erb [hunk 1]
```diff
@@ -6,13 +6,23 @@
...
+    <h2 id="error-summary-title" class="govuk-error-summary__title">
+      There is a problem
+    </h2>
+    <div class="govuk-error-summary__body">
+      <ul class="govuk-list govuk-error-summary__list">
+        <li><%= link_to t('application_form.references.review.more_than_two'), '#references_given' %></li>
+      </ul>
+    </div>
+  </div>
+<% end %>
```
Comment (paulrobertlloyd) [inline app/views/candidate_interface/references/review/show.html.erb:20]: * The error summary should appear before the page title.
* The page title should be prefixed with `Error: `

**Review by paulrobertlloyd — CHANGES_REQUESTED** [review]
In addition to comments:

* Can submit an application with more than 2 references; should show an error on the application review page

* Not quite sure where `app-review-warning` classes are coming from; we now use `govuk-inset-text` with `--warning` and `--error` modifiers for these messages.

File: app/views/candidate_interface/references/review/show.html.erb [hunk 2]
```diff
@@ -1,4 +1,17 @@
...
+
+<% content_for :title, title_with_error_prefix(t('page_titles.references'), @too_many_references) %>
 <% content_for :before_content, govuk_back_link_to(candidate_interface_application_form_path, 'Back to application') %>
```
Comment (davidgisbey) [inline app/views/candidate_interface/references/review/show.html.erb:16]: I think lines 1-12 should be here. It's normally: 

backlink

errors

title

Comment (malcolmbaig) [reply app/views/candidate_interface/references/review/show.html.erb]: updated!

**Review by davidgisbey — APPROVED** [review]
I've tested it locally and it's working really well 👍

**Review by paulrobertlloyd — APPROVED** [review]

**Pull request merged**